File: 0684-redundant-connection/0684-redundant-connection.java
class Solution {
    public boolean dfs(int parent, int src, boolean visited[] , List<List<Integer>> adjList){
        visited[src] = true;
         for(int neightbour : adjList.get(src)) {
              if(!visited[neightbour]){
                 if(dfs(src,neightbour,visited,adjList)){
                     return true;
                 }
              } else if(parent  != neightbour ) {
                  return true;
              }
         }
         return false;  
    }
    public int[] findRedundantConnection(int[][] edges) {
     int n = edges.length;
     List<List<Integer>> adjList = new ArrayList<>();
     for(int i = 0 ;i <= n ;i++) {
        adjList.add(new ArrayList<>());
     }
     int ans[] = new int[2];
     for(int edge[] : edges){
         int u = edge[0];
         int v = edge[1];
         adjList.get(u).add(v);
         adjList.get(v).add(u);
           boolean visited[] = new boolean[n+1];
          if(dfs(-1,u,visited,adjList)){
            ans[0] = u;
            ans[1] = v;
             break;
          }
     }
   return ans;
    }
}

File: 0684-redundant-connection/0684-redundant-connection.cpp
class Solution {
public:
  bool dfs(int parent, int src,vector<int>& visited,vector<vector<int>>& adjList) {
          visited[src] = 1;
       for(auto neighbour : adjList[src]) {
           if(!visited[neighbour]) {
              if(dfs(src,neighbour,visited,adjList)){
                 return 1;
              }
           } else if(parent != neighbour){
                    return 1;
           }
       }
        return 0; 
  }
    vector<int> findRedundantConnection(vector<vector<int>>& edges) {
     int n = edges.size();
      vector<vector<int>>adjList(n+1);
      vector<int>removedEdge;
       for(auto& edge  : edges){
          int u = edge[0], v = edge[1];
        adjList[u].push_back(v);
        adjList[v].push_back(u);
         vector<int>visited(n+1,0);
          if(dfs(-1,u,visited,adjList)){
             removedEdge = {u,v};
             break;
          } 
       }
    return removedEdge;
    }
};